### ISSUE TITLE
Gopls test autocompletion issue

### ISSUE THREAD

**Issue by vadimi** [emacs-lsp/lsp-mode#3699]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [ ] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

gopls offers test method generation autocompletion by typing in `func Tes`. But for some reason after test method is generated the text used for method generation gets copied outside of the method declation, please see this video:

https://user-images.githubusercontent.com/1156106/187053274-d75759a5-87cd-4367-afee-f82494002473.mov

### Steps to reproduce

1. open any go test file (`xxx_test.go`)
2. type in `func Tes` to see the autocompletion popup
3. press Enter 

### Expected behavior

test method should get generated without copying text to the end of the method declaration

### Which Language Server did you use?

lsp-go with gopls

### OS

MacOS

**Comment by vikblom** [comment]
Hi, I have the same issue. 

I restarted gopls with the `-rpc.trace` argument and can see the completion going to the server.

Request:
```
[Trace - 16:49:00.821 PM] Sending request 'textDocument/completion - (3947)'.
Params: {"textDocument":{"uri":"file:///home/viktor/tmp/complete/complete_test.go"},"position":{"line":2,"character":13},"context":{"triggerKind":1}}
```

Response:
```
[Trace - 16:49:00.821 PM] Received response 'textDocument/completion - (3947)' in 0ms.
Result: {"isIncomplete":true,"itemDefaults":{"editRange":{"start":{"line":0,"character":0},"end":{"line":0,"character":0}}},"items":[{"label":"TestSome(t *testing.T)","labelDetails":{},"kind":3,"documentation":"complete the parameter","preselect":true,"sortText":"00000","filterText":"TestSome(t *testing.T)","insertTextFormat":2,"textEdit":{"range":{"start":{"line":2,"character":5},"end":{"line":2,"character":5}},"newText":"TestSome(t *testing.T)"}}]}
```

If I understand LSP correctly then having `start == end` like in `"textEdit":{"range":{"start":{"line":2,"character":5},"end":{"line":2,"character":5}}` means the completion will just insert text.

So could this be a problem in `gopls`?

**Comment by vadimi** [comment]
@vikblom I just installed the newly released `gopls@v0.12.0-pre.1` and the issue if fixed for me! Thanks a lot for taking care of it!
